regparser/api_writer.py

```diff
@@ -123,16 +131,396 @@ def write(self, python_object):
             repo.index.commit(path_parts[-1])
 
 
+class XMLWriteContent:
+
+    def __init__(self, path, layers=None):
+        self.path = path
+        self.layers = layers
+        self.layers['definitions'] = self.extract_definitions()
+        self.appendix_sections = 1 # need to track these manually
+
+        self.caps = [chr(i) for i in range(65, 65 + 26)]
+
+        path_parts = self.path.split('/')
+        dir_path = settings.OUTPUT_DIR + os.path.join(*path_parts[:-1])
+
+        if not os.path.exists(dir_path):
+            os.makedirs(dir_path)
+
+        full_path = settings.OUTPUT_DIR + os.path.join(*path_parts)
+        self.full_path = full_path
+
+    def write(self, tree):
+        if isinstance(tree, Node):
+            print 'writing XML'
+            xml_head = '<?xml version="1.0" encoding="UTF-8"?>\n'
```

Comment:
I don't think you need to specify the XML header (`<?xml?>`) here, `lxml.etree.tostring(..., xml_declaration=True, encoding='UTF-8')` can do it.